test/unit/org/apache/cassandra/db/guardrails/GuardrailAllowFilteringTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class GuardrailAllowFilteringTest extends GuardrailTester
+{
+    @Before
+    public void setupTest()
+    {
+        createTable("CREATE TABLE %s (k int PRIMARY KEY, a int, b int)");
+    }
+
+    private void setGuardrail(boolean allowFilteringEnabled)
+    {
+        guardrails().setAllowFilteringEnabled(allowFilteringEnabled);
+    }
+
+    private boolean getGuardrial()
+    {
+        return guardrails().getAllowFilteringEnabled();
+    }
+
+    @Test
+    public void testAllowFilteringDisabled() throws Throwable
+    {
+        boolean originalState = getGuardrial();
+        setGuardrail(false);
+        assertFails("ALLOW FILTERING statement is prohibited", "SELECT * from %s WHERE a = 5 ALLOW FILTERING");
```

Comment:
Last changes in `GuardrailTester` have changed the order of the parameters (so we can specify multiple warning messages). It would be:
```suggestion
        assertFails("SELECT * from %s WHERE a = 5 ALLOW FILTERING", "ALLOW FILTERING statement is prohibited");
```

Reply:
We could also test with `ALLOW FILTERING` enabled:
```java
setGuardrail(true);
assertValid("SELECT * FROM %s WHERE a = 5 ALLOW FILTERING");
```